lib/config.js

```diff
@@ -1653,17 +1673,25 @@ util.runStrictnessChecks = function (config) {
     _warnOrThrow("NODE_ENV value of '"+NODE_ENV+"' is ambiguous.");
   }
 
-  function _warnOrThrow (msg) {
-    var beStrict = process.env.NODE_CONFIG_STRICT_MODE;
-    var prefix = beStrict ? 'FATAL: ' : 'WARNING: ';
-    var seeURL = 'See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+  function _warnOrThrow (msg, neverThrow) {
+    var prefix = beStrict && !neverThrow ? 'FATAL: ' : 'WARNING: ';
 
     console.error(prefix+msg);
-    console.error(prefix+seeURL);
 
-    // Accept 1 and true as truthy values. When set via process.env, Node.js casts them to strings.
-    if (["true", "1"].indexOf(beStrict) >= 0) {
-      throw new Error(prefix+msg+' '+seeURL);
+    // Unless this is a warning that's never fatal, let the user know about strict mode.
+    if (!neverThrow) {
+      var modeInfo = beStrict ? 'Strict mode is on, aborting.' : 'Strict mode is off, continuing.';
+      var modeMessage =  modeInfo + ' See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+      console.error(prefix+modeMessage);
+      if (beStrict) {
+        throw new Error(prefix+msg+' '+modeMessage);
+      }
+    }
+
+    // If we're not in strict mode or this is a message that's safe to disable, let the user
+    // know that they can turn off warnings altogether
+    if (!beStrict || neverThrow) {
+      console.error(prefix+'To disable this warning, set SUPPRESS_NO_CONFIG_WARNING in the environment.');
```

Comment:
Thank you @lorenwest, I think I see what you mean and misunderstood the original comment by @markstos (I missed the ", there are no config files," clause in the third bullet point). To clarify, we *do* want the variable to suppress the specific warnings, even *if* a default config is present, for the cases of NODE_ENV and NODE_APP_INSTANCE settings that don't match config files, correct? (That's what the original reporter wanted, and that's what I would also need for this to make sense for me.)

In that case, I'll refactor the code accordingly. Keep in mind that we still have three cases (no config at all, no instance config, NODE_ENV points to bad config) that require a user notice about the SUPPRESS environment variable, so maybe we'll just make a tiny helper function for those notices.

We may want to consider, however, whether all of this really makes sense -- if, in future, we want to migrate to a single flag that suppresses all warnings during the initialization process, I would suggest rather preparing for that as a breaking change now (with a new environment variable name SUPPRESS_CONFIG_INIT_WARNINGS or such) since we're already inching up towards it anyway. That category of warnings does not seem so broad to me that this would be a bad idea.

If you agree with that as an alternative approach, I will open a separate ticket for it and you can help me make sure we handle the breaking change in a manner consistent with your release process. If you don't see the point of it, I'll just amend this pull request to handle the NO_CONFIG cases as described above.